src/sql/pg/create.sql

```diff
@@ -0,0 +1,147 @@
+(
+    "NOTIFICATION_STATUS_ID"     INTEGER NOT NULL ,
+    "NOTIFICATION_STATUS_CODE"   CHARACTER VARYING(20) NOT NULL ,
+    "NOTIFICATION_STATUS_DESC"   CHARACTER VARYING(255) NULL ,
+    "DB_CREATED_DTTM"            TIMESTAMP WITH TIME ZONE  DEFAULT  NOW()  NOT NULL ,
+    "DB_UPDATED_DTTM"            TIMESTAMP WITH TIME ZONE  DEFAULT  NOW()  NOT NULL ,
+    "DB_UPDATED_USER"            CHARACTER VARYING(40) DEFAULT  USER  NOT NULL 
+);
+
+CREATE UNIQUE INDEX "XPKLU_NOTIFICATION_STATUS" ON "N_LU_NOTIFICATION_STATUS"
+("NOTIFICATION_STATUS_ID"   ASC);
+
+ALTER TABLE "N_LU_NOTIFICATION_STATUS"
+    ADD CONSTRAINT  "PK_LU_NOTIFICATION_STATE" PRIMARY KEY ("NOTIFICATION_STATUS_ID");
+
+CREATE UNIQUE INDEX "XAK1LU_NOTIFICATION_STATUS" ON "N_LU_NOTIFICATION_STATUS"
+("NOTIFICATION_STATUS_CODE"   ASC);
+
+CREATE TABLE "N_NOTIFICATIONS"
+(
+    "NOTIFICATION_ID"            INTEGER NOT NULL DEFAULT nextval('"N_NOTIFICATION_ID_SEQ"'::regclass) ,
+    "NOTIFICATION_STATUS_ID"     INTEGER NOT NULL REFERENCES "N_LU_NOTIFICATION_STATUS"("NOTIFICATION_STATUS_ID"),
+    "CASE_ID"                    INTEGER NOT NULL REFERENCES "N_CASES"("CASE_ID") ,
+    "NOTIFICATION_TARGET"        CHARACTER VARYING(4000) NULL ,
+    "RETRY_COUNT"                INTEGER NULL ,
+    "NEXT_RETRY_DTTM"            TIMESTAMP WITH TIME ZONE NULL ,
+    "IS_ACTIVE"                  SMALLINT NULL ,
+    "LAST_RETRY_DTTM"            TIMESTAMP WITH TIME ZONE NULL ,
+    "DB_CREATED_DTTM"            TIMESTAMP WITH TIME ZONE  DEFAULT  NOW()  NOT NULL ,
+    "DB_UPDATED_DTTM"            TIMESTAMP WITH TIME ZONE  DEFAULT  NOW()  NOT NULL ,
+    "DB_UPDATED_USER"            CHARACTER VARYING(40) DEFAULT  USER  NOT NULL 
+);
+
+CREATE INDEX "XPKN_NOTIFICATIONS" ON "N_NOTIFICATIONS"
+("NOTIFICATION_ID"   DESC);
+
+CREATE INDEX "XIF1N_NOTIFICATIONS" ON "N_NOTIFICATIONS"
+("CASE_ID"   ASC);
+
+CREATE INDEX "XIF2N_NOTIFICATIONS" ON "N_NOTIFICATIONS"
+("NOTIFICATION_STATUS_ID"   ASC);
+
+--
+--  Triggers and Functions
+--
+CREATE OR REPLACE FUNCTION SYNC_DB_CREATED_DTTM() RETURNS TRIGGER AS $$ BEGIN IF row(NEW.*) IS DISTINCT FROM row(OLD.*) THEN NEW."DB_CREATED_DTTM" = now(); RETURN NEW; ELSE RETURN OLD; END IF; END; $$ language 'plpgsql';
+
+CREATE TRIGGER "N_LU_CASE_STATUS_DTTM" BEFORE UPDATE ON "N_LU_CASE_STATUS"   
+  FOR EACH ROW EXECUTE PROCEDURE   SYNC_DB_CREATED_DTTM();
+
+CREATE TRIGGER "N_NOTIFICATIONS_DTTM" BEFORE UPDATE ON "N_NOTIFICATIONS"   
+  FOR EACH ROW EXECUTE PROCEDURE   SYNC_DB_CREATED_DTTM();
+
+CREATE TRIGGER "N_CASES_DTTM" BEFORE UPDATE ON "N_CASES"   
+  FOR EACH ROW EXECUTE PROCEDURE   SYNC_DB_CREATED_DTTM();
+
+CREATE TRIGGER "N_LU_NOTIFICATION_STATUS_DTTM" BEFORE UPDATE ON "N_LU_NOTIFICATION_STATUS"   
+  FOR EACH ROW EXECUTE PROCEDURE   SYNC_DB_CREATED_DTTM();
+
+--
+-- Initial Lookup values
+--
+
+INSERT INTO "N_LU_CASE_STATUS" ("CASE_STATUS_ID", "CASE_STATUS_CODE", "CASE_STATUS_DESC") VALUES (1, 'proposed', 'This case is proposed');
```

Comment:
if the code was dependent on the LU table for logic and values then it would not be so “brittle”
this also is helpful if future environments (qa, perf, prod) do not assume the surrogate key values and use the real values as persisted in db